addon/mixins/model-records/relationships.js

```diff
@@ -8,46 +8,48 @@ const {
   getOwner,
   Mixin,
   A: emberArray
-} = Ember;
+} = Ember
 
 function relationshipMacro(type) {
   return computed('recordType', 'id', function() {
-
-    return get(this, 'model')._debugInfo().propertyInfo.groups.filter(function(group) {
-      return group.name === type;
-    }).reduce((relationships, group) => {
-      group.properties.forEach((property) => {
-        let records = get(this, `model.${property}`);
-
-        // This might have to be written in such a way
-        // as to provide an observer for 'model.'+property
-        // and push onto the array when that property is available
-        if (!isArray(records)) {
-          records = emberArray([records]);
-        }
-
-        let store = getOwner(this).lookup('store:admin');
-        let constructor = get(this, 'model.constructor');
-        let inverse = constructor.inverseFor(property, store);
-        let meta = constructor.metaForProperty(property);
-
-        pushObject(relationships, {
-          name:    property,
-          records,
-          type:    meta.type,
-          inverse: inverse && inverse.name
-        });
-      });
-
-      return relationships;
-    }, emberArray());
-  });
+    return get(this, 'model')
+      ._debugInfo()
+      .propertyInfo
+      .groups
+      .filter(group => group.name === type)
+      .reduce((relationships, group) => {
+        group.properties.forEach(property => {
+          let records = get(this, `model.${property}`)
+
+          // This might have to be written in such a way
+          // as to provide an observer for 'model.'+property
+          // and push onto the array when that property is available
+          if (!isArray(records)) {
+            records = emberArray([records])
+          }
+
+          const store = getOwner(this).lookup('store:admin')
+          const constructor = get(this, 'model.constructor')
+          const inverse = constructor.inverseFor(property, store)
+          const meta = constructor.metaForProperty(property)
+
+          pushObject(relationships, {
+            name: property,
+            records,
+            type: meta.type,
+            inverse: inverse && inverse.name
+          })
+        })
+
+        return relationships
+      }, emberArray())
```

Comment:
Pretty sure this can be simplified. Here's my attempt

```js
const relationships = get(this, 'model')
...
.filter(...)
.map(group => group.properties)
.reduce((acc, properties) => acc.concat(properties), emberArray())
.map(property => {
  let records = get(this, `model.${property}`)
  if (!isArray(records)) {
    records = emberArray([records])
  }

  const store = getOwner(this).lookup('store:admin')
  const constructor = get(this, 'model.constructor')
  const inverse = constructor.inverseFor(property, store)
  const meta = constructor.metaForProperty(property)

  return { 
    name: property,
    records,
    type: meta.type,
    inverse: inverse && inverse.name
  }
})
```

We can fix this stuff later, it's currently fine for this PR.